### PR TITLE
Azure rm routetable

### PR DESCRIPTION
**Pull request by tomvachon**
##### SUMMARY
Azure Route Table Module

Module does not exist at this time and is needed

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
azure_rm_routetable

##### ANSIBLE VERSION
```
ansible 2.3.1.0
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.14 (default, Jan  6 2018, 12:16:16) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

##### ADDITIONAL INFORMATION
The Ansible Azure modules have **incredible difficulty** running on 2.4 or higher (even most issues on 2.3 are around libraries).  They do need a complete re-look for those reason and this is why I developed on 2.3 (but it shouldn't be 2.3 specific code)

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_routetable.py [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'certified'}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_routetable
+version_added: ""
```
Comment (mattclay) [inline lib/ansible/modules/cloud/azure/azure_rm_routetable.py]: This should be: `version_added: "2.5"`

Comment (tomvachon) [reply lib/ansible/modules/cloud/azure/azure_rm_routetable.py]: Matt, I would but the entire Azure setup doesnt run on 2.5 (the CI tests seem to though, so maybe its good enough?).  Its complaining about the azure python module.  If you have a 2.5 requirements.txt which you know works, I'd be happy to update and test.  I was making integration tests now.

**Comment by tomvachon** [comment]
This is going to fail Shippable because its out of VM quota. Nothing to do with my change...

   "msg": "Error creating or updating virtual machine testvm002 - Azure Error: OperationNotAllowed\nMessage: Operation results in exceeding quota limits of Core. Maximum allowed: 10, Current in use: 10, Additional requested: 1. Please read more about quota increase at http://aka.ms/corequotaincrease."

File: docs/docsite/rst/module_docs/azure_rm_routetable_module.rst [hunk 2]
```diff
@@ -0,0 +1,459 @@
+.. _azure_rm_routetable:
```
Comment (mattclay) [inline docs/docsite/rst/module_docs/azure_rm_routetable_module.rst]: This file shouldn't be part of the PR. It is generated when we build our docs.

Comment (tomvachon) [reply docs/docsite/rst/module_docs/azure_rm_routetable_module.rst]: Removed with newest commit

**Comment by tomvachon** [comment]
Can someone please look at this for 2.5 inclusion?

File: lib/ansible/module_utils/azure_rm_common.py [hunk 3]
```diff
@@ -75,6 +75,8 @@
 CIDR_PATTERN = re.compile(r"(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1"
                           r"[0-9]{2}|2[0-4][0-9]|25[0-5])(/([0-9]|[1-2][0-9]|3[0-2]))")
 
+NEXT_HOP_TYPE = ['VirtualNetworkGateway', 'VnetLocal', 'Internet', 'VirtualAppliance']
```
Comment (zikalino) [inline lib/ansible/module_utils/azure_rm_common.py]: @tomvachon you could move this change to azure_rm_routetable module.
if you don't make changes in common code CI is much faster and simpler, as only your module is tested, and you will also avoid any conflicts with other prs

**Comment by zikalino** [comment]
@tomvachon i will review this module soon, but i can't merge it myself.
also i think right now it will be too late to merge it into 2.5.
in a meanwhile would you be interested in including it in Azure Preview Modules here: https://galaxy.ansible.com/Azure/azure_preview_modules/ ?

**Comment by Fred-sun** [comment]
@tomvachon Do you agree with @zikalino? Could you please help update the PR? Thanks!

**Comment by tomvachon** [comment]
@Fred-sun done, sorry i didnt see that comment

**Comment by Fred-sun** [comment]
@zikalino @mattclay Could you help review this PR? Thanks!

**Comment by Fred-sun** [comment]
@zikalino @mattclay Could you help review this PR? Thanks!

**Comment by Fred-sun** [comment]
@zikalino @mattclay Could you help review this PR? Thanks!

**Comment by Fred-sun** [comment]
@zikalino @mattclay Could you help review this PR? Thanks!

**Comment by Fred-sun** [comment]
@tomvachon Thanks for the contribution, Could you help to resolve the conflicting？ Thanks!

**Comment by yuwzho** [comment]
duplicate_of #41175
